### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by tsequeira-vhs** [xamarin/Xamarin.Forms#11549]
### Description
Shell page parameter navigation does not set parameter for page before OnAppearing is called. In prior releases, the OnAppearing used to be called after the paramters were set allowing us to load the content based on the parameter.

### Steps to Reproduce

1. Use Shell page and add 2 pages
2. Navigate from page 1 to page 2 using GoToAsync("Page2?id=5");
3. Observe that OnAppearing of page 2 is called before the id parameter is set

### Expected Behavior
The parameter set should be called before the OnAppearing is called

### Actual Behavior
OnAppearing of page 2 is called before the id parameter is set

### Basic Information

- Version with issue: Xamarin.Forms 4.7.0.1080
- Last known good version: Xamarin.Forms 4.6.x and versions below

**Comment by tsequeira-vhs** [comment]
Will this issue be fixed in 5.0.0?

**Comment by elcainas** [comment]
I tested in Xamarin.Forms 4.6.x , 4.8.x  same issue

**Comment by tsequeira-vhs** [comment]
@samhouts Does this mean that this issue will no longer be fixed?

**Comment by attilavlacsil** [comment]
With 4.8.0.1687 I have the same issue, but the actual behavior happens only if I navigate to the page for the first time, any subsequent navigation works as expected.

If I have a similar page to track constructor, query property and OnAppearing call order

```csharp
[QueryProperty(nameof(QueryData), nameof(QueryData))]
public partial class ShellPage : ContentPage
{
    private readonly ShellPageViewModel viewModel;

    public string QueryData
    {
        set
        {
            Trace.WriteLine("ShellPage.QueryData setter");

            viewModel.QueryData = value;
        }
    }

    public ShellPage()
    {
        Trace.WriteLine("ShellPage ctor");

        InitializeComponent();
        BindingContext = viewModel = new ShellPageViewModel();
    }

    protected override void OnAppearing()
    {
        Trace.WriteLine("ShellPage.OnAppearing");

        base.OnAppearing();
    }
}
```

Navigating to the page for the first time outputs

```
[0:] ShellPage ctor
[0:] ShellPage.OnAppearing
[0:] ShellPage.QueryData setter
```

Any subsequent navigation outputs

```
[0:] ShellPage.QueryData setter
[0:] ShellPage.OnAppearing
```

As mentioned, the order must be consistent for any given navigation.

**Comment by archergod** [comment]
I am having same Issue with 5.0.0.2012

**Comment by andrybany** [comment]
I have also had this problem for some time. 
A little while ago I just discovered that it can be remedied like this:

```C#
protected async override void OnAppearing()
{
    base.OnAppearing();
    await Task.Yield();
    //DO some stuff with QueryProperty populated 
}
```

**Comment by K-Ahmad** [comment]
I am having same problem 5.0.0.2478

**Comment by janseris** [comment]
@PureWeen  Hello? This is still a problem in MAUI 7.0.81. It works (ApplyQueryAttributes called before OnAppearing) either only first time or never (it's not determistic).

**Comment by Scott-McEachern-SRT** [comment]
Still an issue with MAUI. The first time the page loads the query properties are not populated when the OnAppearing is fired.

**Comment by PureWeen** [comment]
> Still an issue with MAUI. The first time the page loads the query properties are not populated when the OnAppearing is fired.

Is there an issue on the MAUI repository for this?

**Comment by plppp2001** [comment]
Yes, I just noticed this happens sometimes for me 30% of the time OnAppearing gets called before ApplyQueryAttributes.

**Comment by janseris** [comment]
> > Still an issue with MAUI. The first time the page loads the query properties are not populated when the OnAppearing is fired.
> 
> Is there an issue on the MAUI repository for this?

Should it be so that it is solved? Shell is for both MAUI  and Xamarin or does it differ?

**Comment by plppp2001** [comment]
Is there Maui bug for this out there?
